### PR TITLE
Enable user to set the custom vm identity provider for credential authentication

### PR DESCRIPTION
**Pull request by williamaronli**
Please provide a description for what this PR is for.
ref: https://github.com/istio/istio/issues/27947
And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: security/pkg/credentialfetcher/plugin/gce.go [hunk 1]
```diff
@@ -80,5 +84,5 @@ func (p *GCEPlugin) GetType() string {
 // GetIdentityProvider returns the name of the identity provider that can authenticate the workload credential.
 // GCE idenity provider is named "GoogleComputeEngine".
```
Comment (howardjohn) [inline security/pkg/credentialfetcher/plugin/gce.go:85]: Why would the GCE credential fetcher ever not have GoogleComputeEngine as the IDP?

**Review by howardjohn — DISMISSED** [review]
Can you add some more context to the PR? I am not sure the background on this.

File: security/pkg/credentialfetcher/plugin/gce.go (hunk 1 above)
Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: sometimes we support the other credential fetcher like AWS. so we need to enable the customer to set the customer identity provider when setup vm

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -147,6 +147,8 @@ var (
 	useTokenForCSREnv   = env.RegisterBoolVar("USE_TOKEN_FOR_CSR", false, "CSR requires a token").Get()
 	credFetcherTypeEnv  = env.RegisterStringVar("CREDENTIAL_FETCHER_TYPE", "",
 		"The type of the credential fetcher. Currently supported types include GoogleComputeEngine").Get()
+	credIdentityProvider = env.RegisterStringVar("CREDENTIAL_IDENTITY_PROVIDER", "GoogleComputeEngine",
+		"The identity provider for CREDENTIAL. Currently Default supported identity provider is GoogleComputeEngine").Get()
```
Comment (JimmyCYJ) [inline pilot/cmd/pilot-agent/main.go:151]: ...provider for credential. Currently default...

File: releasenotes/notes/27879.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: release-notes/v2
+kind: feature
+area: security
+issue:
+- 27879
```
Comment (JimmyCYJ) [inline releasenotes/notes/27879.yaml:5]: This is a issue ID not PR ID.

File: security/pkg/credentialfetcher/plugin/gce.go (hunk 1 above)
Comment (howardjohn) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: Can you explain what it means to have
```
CredentialFetcher=AWS
IDP=GCE
```
?

Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: IDP is used in IDNS mapping. and when CredentialFetcher=AWS, IDP should also be AWS

Comment (howardjohn) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: if CredentialFetcher must always match IDP then why is it configurable?

Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: user can set multiple identity provider even if VM is in AWS.

Comment (howardjohn) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: Can you please explain why and how a user is expected to interact with this? If I cannot figure it out, our users will not either

Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: the identity provider can authenticate the workload credential and example for GCE idenity provider is named "GoogleComputeEngine".

Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: @howardjohn  this is required for future VM support for different IDNS mapping for different platform

Comment (williamaronli) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: you can reference to this issue: 

Currently we hardcoded the Identity provider(GoogleComputeEngine) which is the identifier to find the correct IDNS mapping and credential Fetcher when setting up the VM. To enable the customed identity to be used in the, we choose to add a new environment variable CREDENTIAL_IDENTITY_PROVIDER to transmit this identity to enable the istio agent to find :

```
credIdentityProvider = env.RegisterStringVar("CREDENTIAL_IDENTITY_PROVIDER", "GoogleComputeEngine",
"The identity provider for CREDENTIAL. Currently Default supported identity provider is GoogleComputeEngine").Get()
```

Comment (liminw) [reply security/pkg/credentialfetcher/plugin/gce.go:85]: GCE or AWS is probably not a good example. I am thinking of on-prem VM use case, where I could have a credential fetcher reading a token from a local file, which has "filereader" CredentialFetcher type. All my on-prem VMs may be configured with the same "filereader" credentialFetcher type, but the VMs may use different IdentityProviders which have different public key endpoints. 

Credential fetcher decides how a credential is fetched for a workload. IdentityProvider specifies how the credential is authenticated. They serve different purposes. In many cases, the two match. But sometimes they are different. 

Does it make sense?

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**